### PR TITLE
feat: added sql support for domains

### PR DESCRIPTION
**Pull request by markusahlstrand**
Ok, got 2 things in here.. The main thing is adding the basic sql models to handle domains. Currently it's just the sql stuff so we know which tenant has which custom domains connected to it. The next step is to integrate with the cloudflare api so we can manage the custom domains there.

I also did some prepping to make it possible to deploy this to prod. I added a small script to create the infra needed and added the config variables to the toml file.

### CODE REVIEW

File: src/migrate/migrations/2023-08-15T09:17:35_domains.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { Database } from "../../types";
+
+export async function up(db: Kysely<Database>): Promise<void> {
+  await db.schema
+    .createTable("domains")
+    .addColumn("id", "varchar(255)", (col) => col.notNull().primaryKey())
+    .addColumn("tenant_id", "varchar(255)", (col) =>
+      col.references("tenants.id").onDelete("cascade").notNull(),
+    )
+    .addColumn("domain", "varchar(255)", , (col) => col.notNull().unique())
```
Comment (danielgent) [inline src/migrate/migrations/2023-08-15T09:17:35_domains.ts:11]: @markusahlstrand prettier is complaining BUT it's not a formatting issue for once! :smile:

Comment (danielgent) [inline src/migrate/migrations/2023-08-15T09:17:35_domains.ts:11]:
```suggestion
    .addColumn("domain", "varchar(255)", (col) => col.notNull().unique())
```

File: src/routes/tsoa/domains.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    @Request() request: RequestWithContext,
+    @Path("id") id: string,
+    @Path("tenantId") tenantId: string,
+  ): Promise<string> {
+    const { env } = request.ctx;
+
+    const db = getDb(env);
+    await db
+      .deleteFrom("domains")
+      .where("domains.tenantId", "=", tenantId)
```
Comment (danielgent) [inline src/routes/tsoa/domains.ts:96]: Will multiple domains have the same `id`? I don't think so

There's no reason to do `.where("domains.tenantId", "=", tenantId)` right?

In the functions below you're not doing this.  The id is the primary key so if `.where("domains.id", "=", id)` then we have the right one

*BUT* I think here it's an extra check? is it a security thing? I don't think so as admins are admins

**Review by danielgent — APPROVED** [review]
Some questions but nothing I see is an issue :+1:

File: src/routes/tsoa/domains.ts (hunk 2 above)
Comment (markusahlstrand) [reply src/routes/tsoa/domains.ts:96]: It is true that domains are unique, but if we don't check for the tenant the user could delete a domain from another users tenant :)

**Pull request merged**